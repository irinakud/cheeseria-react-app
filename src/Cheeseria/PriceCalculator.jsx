import React, { useState } from 'react';
import calculatePrice from './calculatePrice';

export const PriceCalculator = ({pricePerKilo}) => {
    const [totalPrice, setTotalPrice] = useState(0);

    const onAmountChange = (amount) => {
        setTotalPrice(calculatePrice(amount, pricePerKilo));
    }

    return (
        <div>
            <div>
                Enter amount (in kg): 
                <input onChange={e => onAmountChange(e.target.value)}/>
            </div>
            <div>Total Price: {totalPrice}</div>                
        </div>
    );
}
export default PriceCalculator;