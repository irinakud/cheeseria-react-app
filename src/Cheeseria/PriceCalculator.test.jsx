import { render } from "@testing-library/react";
import {PriceCalculator} from "./PriceCalculator";

describe("PriceCalculator", () => {
  it("can render", () => {
      expect.hasAssertions();

      const { baseElement } = render(<PriceCalculator/>)
    
      expect(baseElement).toMatchSnapshot();
  });

  // TODO: more tests to add
});
