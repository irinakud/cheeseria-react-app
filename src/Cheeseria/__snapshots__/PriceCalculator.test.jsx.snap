// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceCalculator can render 1`] = `
<body>
  <div>
    <div>
      <div>
        Enter amount (in kg):
        <input />
      </div>
      <div>
        Total Price: 
        0
      </div>
    </div>
  </div>
</body>
`;
